packages/dds/tree/src/test/simple-tree/api/schemaFromSimple.spec.ts

```diff
@@ -69,4 +74,43 @@ describe("schemaFromSimple", () => {
 			roundtrip(testTreeSchema);
 		});
 	});
+
+	describe("compatibility fields", () => {
+		it("handles allowUnknownOptionalFields", () => {
+			const root = HasUnknownOptionalFields;
+			const simpleSchema = getSimpleSchema(root);
+			const simpleObjectSchema = simpleSchema.definitions.get(
+				"test.hasUnknownOptionalFields",
+			) as SimpleObjectNodeSchema;
+			assert.equal(simpleObjectSchema.allowUnknownOptionalFields, true);
+
+			const viewSchema = generateSchemaFromSimpleSchema(simpleSchema);
+			const objectViewSchema = viewSchema.definitions.get(
+				"test.hasUnknownOptionalFields",
+			) as SimpleObjectNodeSchema;
+			assert.equal(objectViewSchema.allowUnknownOptionalFields, true);
+		});
+
+		it("handles staged allowed types", () => {
+			const root = HasStagedAllowedTypes;
```

Comment:
I think rather than important some existing schema, which is external to this test (and hard to know what it is when reading this test), and is needlessly wrapped up in an object schema, iy would be better to just inline a specific schema here. Maybe:

```
SchemaFactoryBeta.types([
		SchemaFactory.number,
		SchemaFactory.staged(SchemaFactory.string),
	])
```

Then you can assert both cases, that number is staged and that string is not to make sure we don't accidently stage everything.